---
title: "Intro to Transcriptomics"
author: Nicole Black, Wade Boohar, Kayla Xu
date: 07/17/22
updated: 09/26/23
---

***Deliverables***
-Upload this R Notebook to your GitHub and submit the link to your Repo on BB.
-Include any graphs or figures created in this assignment in the folder with your R notebook with descriptive file names.

Since this is an optional partner activity, it is okay if your answers are the same as your partnerâ€™s as long as everyone understands it and could explain it in their own words if asked. Each person must individually push their code to Github. *At the top of your R Notebook, write the name of you and your partner(s) as a comment.*

***Complete the following coding activity and answer any following questions as comments in your R Notebook***

In Transcriptomics_Tutorial.Rmd, you learned how to manipulate the SummarizedExperiment data structure, turn it into more readable dataframes, complete basic analysis using those dataframes, and performed a differential expression analysis based on age. Before saving the rna_counts, rna_clinical, and rna_genes however, we masked out any sample that was not a tumor. In this assignment, you will need to reinstantiate the rna_se object and then perform differential expression analysis based on tumor status.

*Pre-Assignment*
Use knitr function to set your working directory to your analysis_data folder in qbio_490_name.
```{r}
    knitr::opts_knit$set(root.dir = normalizePath("/Users/lantrieu1964/Documents/GitHub/qbio_490_LanTrieu/analysis_data")) 
```

Load all necessary packages and instantiate the rna_se object
```{r}
if (!require("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install(version = "3.17")
library(BiocManager)
if (!require("TCGAbiolinks", quietly = TRUE))
  BiocManager::install("TCGAbiolinks")
library(TCGAbiolinks)
if (!require("maftools", quietly = TRUE))
  BiocManager::install("maftools")
library(maftools)
if (!require("DESeq2", quietly = TRUE))
  BiocManager::install("DESeq2")
library(DESeq2)
if (!require("EnhancedVolcano", quietly = TRUE))
  BiocManager::install("EnhancedVolcano")
library(EnhancedVolcano)

rna_query <- GDCquery(project ="TCGA-BRCA",
                      data.category = "Transcriptome Profiling",
                      data.type = "Gene Expression Quantification",
                      workflow.type = "STAR - Counts")
#GDCdownload(rna_query)

rna_se <- GDCprepare(rna_query)
```


*1*
Perform all data cleaning and label manipulation steps (hint: applying colnames and rownames to make the dataframes more readable) except for masking out the patient with an NA for age and the "Solid Tissue Normal" sample types.
```{r}
rna_clinical <- rna_se@colData
rna_clinical <- as.data.frame(rna_clinical)
treatments_mask <- ifelse(colnames(rna_clinical) == 'treatments',F,T)
rna_clinical <- rna_clinical[,treatments_mask]
primary_site_mask <- ifelse(colnames(rna_clinical) == 'primary_site',F,T)
rna_clinical <- rna_clinical[,primary_site_mask]
disease_type_mask <- ifelse(colnames(rna_clinical) == 'disease_type',F,T)
rna_clinical <- rna_clinical[,disease_type_mask]
rna_genes <- rna_se@rowRanges@elementMetadata
rna_genes <- as.data.frame(rna_genes)
rna_counts <- rna_se@assays@data$unstranded
rna_counts <- as.data.frame(rna_counts)
rna_clinical$age_category <- ifelse(rna_clinical$age_at_index <=58, 'young','old')
age_na_mask <- !is.na(rna_clinical$age_category)
rna_clinical <- rna_clinical[!is.na(rna_clinical$age_category),]
rna_counts <- rna_counts[,!is.na(rna_clinical$age_category)]
rownames(rna_genes)
colnames(rna_genes)
rownames(rna_genes) <- rna_genes$gene_id
```
*2*
Determine and define your experimental design for the differential expression analysis. Control for covariance of age, gender, OR vital_status. Manipulate those columns so that they are ready for differential expression analysis (hint: what kind of variables are they? what data type are they by default? do you need to handle unknown values?)
```{r}
rna_clinical <- rna_se@colData
rna_clinical <- as.data.frame(rna_clinical)
treatments_mask <- ifelse(colnames(rna_clinical) == 'treatments',F,T)
rna_clinical <- rna_clinical[,treatments_mask]
primary_site_mask <- ifelse(colnames(rna_clinical) == 'primary_site',F,T)
rna_clinical <- rna_clinical[,primary_site_mask]
disease_type_mask <- ifelse(colnames(rna_clinical) == 'disease_type',F,T)
rna_clinical <- rna_clinical[,disease_type_mask]
rna_genes <- rna_se@rowRanges@elementMetadata
rna_genes <- as.data.frame(rna_genes)
rna_counts <- rna_se@assays@data$unstranded
rna_counts <- as.data.frame(rna_counts)
rna_clinical$age_category <- ifelse(rna_clinical$age_at_index <=58, 'young','old')
age_na_mask <- !is.na(rna_clinical$age_category)
rna_clinical <- rna_clinical[!is.na(rna_clinical$age_category),]
rna_counts <- rna_counts[,!is.na(rna_clinical$age_category)]
rownames(rna_genes) <- rna_genes$gene_id
rownames(rna_counts) <- rna_genes$gene_id
colnames(rna_counts) <- rownames(rna_clinical)
```


*3*
Perform the differential expression analysis, create a volcano plot, and interpret what genes from each of the 6 sections means (assuming thresholds of -log10padj = 2 and log2FoldChange = 1)
```{r}
rna_clinical$age_category <- factor(rna_clinical$age_category)
dds <- DESeqDataSetFromMatrix(countData = rna_counts,
                              colData = rna_clinical,
                              design = ~ vital_status + gender + age_category)
dds_obj <- DESeq(dds)
resultsNames(dds_obj) 
head(results)
results <- results(dds_obj, format = "DataFrame", contrast = c("age_category", "old", "young"))
results <- data.frame(rna_genes$gene_name, rownames(results), results$log2FoldChange, results$pvalue, results$padj, -log10(results$padj))
colnames(results) <- c("gene_name", "gene_id", "log2FoldChange", "pvalue", "padj", "-log10(padj)")
row.names(results) <- rna_genes$gene_id
par(mar = c(0.5, 1, 1, 0.5))
EnhancedVolcano(results,
                lab = results$gene_name,
                x = 'log2FoldChange',
                y = 'pvalue',
                titleLabSize = 10,
                subtitleLabSize = 6,
                captionLabSize = 5,
                axisLabSize = 10,
                legendPosition = "right",
                legendLabSize = 10,
                labSize = 3,
                pointSize = 1,
                gridlines.major = FALSE,
                gridlines.minor = FALSE
)
```

top-right genes: Significant, up-regulated genes
bottom-right genes: Insignificant, up-regulated genes
top-left genes: Significant, down-regulated genes
bottom-left genes: Insignificant, down-regulated genes
top-middle genes: Significant genes (neither up, nor down-regulated)
bottom-middle genes: Insignificant genes (neither up, nor down-regulated genes)

Save the picture of the volcano plot (using either ggsave() or right clicking and manually downloading the image and push this .Rmd and the image to GitHub)