{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a53acaad",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.chdir('/Users/lantrieu1964/Documents/GitHub/qbio_490_LanTrieu/analysis_data')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3274cad7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Your version of cptac (1.1.2) is out-of-date. Latest is 1.5.7. Please run 'pip install --upgrade cptac' to update it. (/Users/lantrieu1964/miniconda3/envs/qbio_490_LanTrieu/lib/python3.9/threading.py, line 917)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Description</th>\n",
       "      <th>Data reuse status</th>\n",
       "      <th>Publication link</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Dataset name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Brca</th>\n",
       "      <td>breast cancer</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/33212010/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ccrcc</th>\n",
       "      <td>clear cell renal cell carcinoma (kidney)</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/31675502/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Colon</th>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/31031003/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Endometrial</th>\n",
       "      <td>endometrial carcinoma (uterine)</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/32059776/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gbm</th>\n",
       "      <td>glioblastoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/33577785/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hnscc</th>\n",
       "      <td>head and neck squamous cell carcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/33417831/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lscc</th>\n",
       "      <td>lung squamous cell carcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/34358469/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Luad</th>\n",
       "      <td>lung adenocarcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/32649874/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ovarian</th>\n",
       "      <td>high grade serous ovarian cancer</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/27372738/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pdac</th>\n",
       "      <td>pancreatic ductal adenocarcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/34534465/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UcecConf</th>\n",
       "      <td>endometrial confirmatory carcinoma</td>\n",
       "      <td>password access only</td>\n",
       "      <td>unpublished</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GbmConf</th>\n",
       "      <td>glioblastoma confirmatory</td>\n",
       "      <td>password access only</td>\n",
       "      <td>unpublished</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           Description     Data reuse status  \\\n",
       "Dataset name                                                                   \n",
       "Brca                                     breast cancer       no restrictions   \n",
       "Ccrcc         clear cell renal cell carcinoma (kidney)       no restrictions   \n",
       "Colon                                colorectal cancer       no restrictions   \n",
       "Endometrial            endometrial carcinoma (uterine)       no restrictions   \n",
       "Gbm                                       glioblastoma       no restrictions   \n",
       "Hnscc            head and neck squamous cell carcinoma       no restrictions   \n",
       "Lscc                      lung squamous cell carcinoma       no restrictions   \n",
       "Luad                               lung adenocarcinoma       no restrictions   \n",
       "Ovarian               high grade serous ovarian cancer       no restrictions   \n",
       "Pdac                  pancreatic ductal adenocarcinoma       no restrictions   \n",
       "UcecConf            endometrial confirmatory carcinoma  password access only   \n",
       "GbmConf                      glioblastoma confirmatory  password access only   \n",
       "\n",
       "                                       Publication link  \n",
       "Dataset name                                             \n",
       "Brca          https://pubmed.ncbi.nlm.nih.gov/33212010/  \n",
       "Ccrcc         https://pubmed.ncbi.nlm.nih.gov/31675502/  \n",
       "Colon         https://pubmed.ncbi.nlm.nih.gov/31031003/  \n",
       "Endometrial   https://pubmed.ncbi.nlm.nih.gov/32059776/  \n",
       "Gbm           https://pubmed.ncbi.nlm.nih.gov/33577785/  \n",
       "Hnscc         https://pubmed.ncbi.nlm.nih.gov/33417831/  \n",
       "Lscc          https://pubmed.ncbi.nlm.nih.gov/34358469/  \n",
       "Luad          https://pubmed.ncbi.nlm.nih.gov/32649874/  \n",
       "Ovarian       https://pubmed.ncbi.nlm.nih.gov/27372738/  \n",
       "Pdac          https://pubmed.ncbi.nlm.nih.gov/34534465/  \n",
       "UcecConf                                    unpublished  \n",
       "GbmConf                                     unpublished  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Import CPTAC & Examine Data Sets Available\n",
    "import cptac\n",
    "cptac.list_datasets()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "aeb53aa2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Below are the dataframes contained in this dataset and their dimensions:\n",
      "\n",
      "clinical\n",
      "\t194 rows\n",
      "\t171 columns\n",
      "CNV\n",
      "\t110 rows\n",
      "\t19285 columns\n",
      "followup\n",
      "\t352 rows\n",
      "\t27 columns\n",
      "medical_history\n",
      "\t370 rows\n",
      "\t4 columns\n",
      "methylation\n",
      "\t107 rows\n",
      "\t15885 columns\n",
      "phosphoproteomics\n",
      "\t194 rows\n",
      "\t81550 columns\n",
      "phosphoproteomics_gene\n",
      "\t194 rows\n",
      "\t6127 columns\n",
      "proteomics\n",
      "\t194 rows\n",
      "\t11710 columns\n",
      "somatic_mutation\n",
      "\t8350 rows\n",
      "\t3 columns\n",
      "transcriptomics\n",
      "\t185 rows\n",
      "\t19275 columns\n"
     ]
    }
   ],
   "source": [
    "#Load & Download CCRCC Dataset from CPTAC\n",
    "cptac.download(dataset = \"Ccrcc\")\n",
    "ccrcc = cptac.Ccrcc()\n",
    "ccrcc.list_data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2f7abc05",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/lantrieu1964/miniconda3/envs/qbio_490_LanTrieu/lib/python3.9/site-packages/pandas/core/internals/blocks.py:366: RuntimeWarning: divide by zero encountered in log2\n",
      "  result = func(self.values, **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Sample_Tumor_Normal</th>\n",
       "      <th>tumor/normal</th>\n",
       "      <th>gender</th>\n",
       "      <th>age</th>\n",
       "      <th>height_in_cm</th>\n",
       "      <th>height_in_inch</th>\n",
       "      <th>weight_in_kg</th>\n",
       "      <th>weight_in_lb</th>\n",
       "      <th>BMI</th>\n",
       "      <th>race</th>\n",
       "      <th>...</th>\n",
       "      <th>histologic_type_of_normal_tissue</th>\n",
       "      <th>slide_is_free_of_tumor</th>\n",
       "      <th>consistent_with_local_pathology_report</th>\n",
       "      <th>findings_not_consistent_with_local_pathology_report</th>\n",
       "      <th>weight_in_mg</th>\n",
       "      <th>minutes_clamp_1_to_collection</th>\n",
       "      <th>minutes_clamp_2_to_collection</th>\n",
       "      <th>minutes_collection_to_frozen</th>\n",
       "      <th>consistent_with_diagnostic_report</th>\n",
       "      <th>patient_medications</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>72</td>\n",
       "      <td>170.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>22.80</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>484.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rivaroxaban,tylenol,Aspirin,glycolax,colace,zocor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>30</td>\n",
       "      <td>177.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>236.0</td>\n",
       "      <td>34.15</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>575.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Rivaroxaban,Esomeprazole ,Tramadol</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Female</td>\n",
       "      <td>65</td>\n",
       "      <td>163.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>28.23</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>212.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Levothyroxine Sodium,Cyproheptadine HCL,Citrac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>49</td>\n",
       "      <td>175.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>116.0</td>\n",
       "      <td>256.0</td>\n",
       "      <td>37.88</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>675.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28</td>\n",
       "      <td>Yes</td>\n",
       "      <td>ibuprofen,Norco,miralax</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00088</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>72</td>\n",
       "      <td>178.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>229.0</td>\n",
       "      <td>32.82</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>306.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Lisinopril,Victoza,Tamsulosin,Simvastatin,Quet...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>69</td>\n",
       "      <td>170.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>25.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>303.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>20</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Rosuvastatinum,Humulin N,fenofibratum,torasemi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>69</td>\n",
       "      <td>176.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>192.0</td>\n",
       "      <td>28.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>129.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Ramipril,Bisoprolol</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>51</td>\n",
       "      <td>173.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>231.0</td>\n",
       "      <td>35.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>228.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29</td>\n",
       "      <td>Yes</td>\n",
       "      <td>metoprolol,torasemidum,Methyldopum,nitrendypin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>58</td>\n",
       "      <td>172.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>165.0</td>\n",
       "      <td>25.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>287.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22</td>\n",
       "      <td>Yes</td>\n",
       "      <td>fraxiparine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>TN</td>\n",
       "      <td>Male</td>\n",
       "      <td>47</td>\n",
       "      <td>183.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>212.0</td>\n",
       "      <td>28.75</td>\n",
       "      <td>White</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>144.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>98 rows Ã— 171 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       Sample_Tumor_Normal tumor/normal  gender age  height_in_cm  \\\n",
       "Patient_ID                                                              \n",
       "C3L-00004                Tumor           TN    Male  72         170.0   \n",
       "C3L-00010                Tumor           TN    Male  30         177.0   \n",
       "C3L-00026                Tumor           TN  Female  65         163.0   \n",
       "C3L-00079                Tumor           TN    Male  49         175.0   \n",
       "C3L-00088                Tumor           TN    Male  72         178.0   \n",
       "...                        ...          ...     ...  ..           ...   \n",
       "C3N-01646                Tumor           TN    Male  69         170.0   \n",
       "C3N-01648                Tumor           TN    Male  69         176.0   \n",
       "C3N-01649                Tumor           TN    Male  51         173.0   \n",
       "C3N-01651                Tumor           TN    Male  58         172.0   \n",
       "C3N-01808                Tumor           TN    Male  47         183.0   \n",
       "\n",
       "Name        height_in_inch  weight_in_kg  weight_in_lb    BMI   race  ...  \\\n",
       "Patient_ID                                                            ...   \n",
       "C3L-00004             67.0          66.0         145.0  22.80  White  ...   \n",
       "C3L-00010             70.0         107.0         236.0  34.15  White  ...   \n",
       "C3L-00026             64.0          75.0         165.0  28.23  White  ...   \n",
       "C3L-00079             69.0         116.0         256.0  37.88  White  ...   \n",
       "C3L-00088             70.0         104.0         229.0  32.82  White  ...   \n",
       "...                    ...           ...           ...    ...    ...  ...   \n",
       "C3N-01646             67.0          75.0         165.0  25.00    NaN  ...   \n",
       "C3N-01648             69.0          87.0         192.0  28.00    NaN  ...   \n",
       "C3N-01649             68.0         105.0         231.0  35.00    NaN  ...   \n",
       "C3N-01651             68.0          75.0         165.0  25.00    NaN  ...   \n",
       "C3N-01808             72.0          96.0         212.0  28.75  White  ...   \n",
       "\n",
       "Name       histologic_type_of_normal_tissue slide_is_free_of_tumor  \\\n",
       "Patient_ID                                                           \n",
       "C3L-00004                               NaN                    NaN   \n",
       "C3L-00010                               NaN                    NaN   \n",
       "C3L-00026                               NaN                    NaN   \n",
       "C3L-00079                               NaN                    NaN   \n",
       "C3L-00088                               NaN                    NaN   \n",
       "...                                     ...                    ...   \n",
       "C3N-01646                               NaN                    NaN   \n",
       "C3N-01648                               NaN                    NaN   \n",
       "C3N-01649                               NaN                    NaN   \n",
       "C3N-01651                               NaN                    NaN   \n",
       "C3N-01808                               NaN                    NaN   \n",
       "\n",
       "Name       consistent_with_local_pathology_report  \\\n",
       "Patient_ID                                          \n",
       "C3L-00004                                     Yes   \n",
       "C3L-00010                                     Yes   \n",
       "C3L-00026                                     Yes   \n",
       "C3L-00079                                     Yes   \n",
       "C3L-00088                                     Yes   \n",
       "...                                           ...   \n",
       "C3N-01646                                     Yes   \n",
       "C3N-01648                                     Yes   \n",
       "C3N-01649                                     Yes   \n",
       "C3N-01651                                     Yes   \n",
       "C3N-01808                                     Yes   \n",
       "\n",
       "Name       findings_not_consistent_with_local_pathology_report  weight_in_mg  \\\n",
       "Patient_ID                                                                     \n",
       "C3L-00004                                                 NaN          484.0   \n",
       "C3L-00010                                                 NaN          575.0   \n",
       "C3L-00026                                                 NaN          212.0   \n",
       "C3L-00079                                                 NaN          675.0   \n",
       "C3L-00088                                                 NaN          306.0   \n",
       "...                                                       ...            ...   \n",
       "C3N-01646                                                 NaN          303.0   \n",
       "C3N-01648                                                 NaN          129.0   \n",
       "C3N-01649                                                 NaN          228.0   \n",
       "C3N-01651                                                 NaN          287.0   \n",
       "C3N-01808                                                 NaN          144.0   \n",
       "\n",
       "Name       minutes_clamp_1_to_collection minutes_clamp_2_to_collection  \\\n",
       "Patient_ID                                                               \n",
       "C3L-00004                            NaN                           NaN   \n",
       "C3L-00010                            NaN                           NaN   \n",
       "C3L-00026                           13.0                           NaN   \n",
       "C3L-00079                            NaN                           NaN   \n",
       "C3L-00088                            NaN                           NaN   \n",
       "...                                  ...                           ...   \n",
       "C3N-01646                            NaN                           NaN   \n",
       "C3N-01648                            NaN                           NaN   \n",
       "C3N-01649                            NaN                           NaN   \n",
       "C3N-01651                            NaN                           NaN   \n",
       "C3N-01808                            NaN                           NaN   \n",
       "\n",
       "Name       minutes_collection_to_frozen consistent_with_diagnostic_report  \\\n",
       "Patient_ID                                                                  \n",
       "C3L-00004                            21                               Yes   \n",
       "C3L-00010                            17                               Yes   \n",
       "C3L-00026                            13                               Yes   \n",
       "C3L-00079                            28                               Yes   \n",
       "C3L-00088                            29                               Yes   \n",
       "...                                 ...                               ...   \n",
       "C3N-01646                            20                               Yes   \n",
       "C3N-01648                             8                               Yes   \n",
       "C3N-01649                            29                               Yes   \n",
       "C3N-01651                            22                               Yes   \n",
       "C3N-01808                            29                               Yes   \n",
       "\n",
       "Name                                      patient_medications  \n",
       "Patient_ID                                                     \n",
       "C3L-00004   rivaroxaban,tylenol,Aspirin,glycolax,colace,zocor  \n",
       "C3L-00010                  Rivaroxaban,Esomeprazole ,Tramadol  \n",
       "C3L-00026   Levothyroxine Sodium,Cyproheptadine HCL,Citrac...  \n",
       "C3L-00079                             ibuprofen,Norco,miralax  \n",
       "C3L-00088   Lisinopril,Victoza,Tamsulosin,Simvastatin,Quet...  \n",
       "...                                                       ...  \n",
       "C3N-01646   Rosuvastatinum,Humulin N,fenofibratum,torasemi...  \n",
       "C3N-01648                                 Ramipril,Bisoprolol  \n",
       "C3N-01649   metoprolol,torasemidum,Methyldopum,nitrendypin...  \n",
       "C3N-01651                                         fraxiparine  \n",
       "C3N-01808                                                 NaN  \n",
       "\n",
       "[98 rows x 171 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "clinical_data = ccrcc.get_clinical()\n",
    "\n",
    "#Create Mask for RNA Data\n",
    "rna_data = ccrcc.get_transcriptomics()\n",
    "rna_mask = rna_data.index.isin(clinical_data.index)\n",
    "rna_data = rna_data[rna_mask]\n",
    "rna_data = np.log2(rna_data)\n",
    "rna_data = rna_data.dropna(axis = 1)\n",
    "rna_data = rna_data.T.loc[~rna_data.T.index.duplicated(), :].T\n",
    "\n",
    "#Create Mask for Protein Data\n",
    "protein_data = ccrcc.get_proteomics()\n",
    "protein_data.columns = protein_data.columns.get_level_values(0) \n",
    "protein_mask = protein_data.index.isin(clinical_data.index)\n",
    "protein_data = protein_data[protein_mask]\n",
    "protein_data = protein_data.dropna(axis = 1)\n",
    "protein_data = protein_data.T.loc[~protein_data.T.index.duplicated(), :].T\n",
    "\n",
    "#Create Masks for Stage I-III Tumors\n",
    "stage_mask = np.where(clinical_data.tumor_stage_pathological != \"Stage IV\", True, False)\n",
    "clinical_data = clinical_data[stage_mask]\n",
    "clinical_data = clinical_data[clinical_data.tumor_stage_pathological.notna()]\n",
    "clinical_data.tumor_stage_pathological.unique()\n",
    "clinical_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "15cf516a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top 5 RNA Genes:\n",
      "Name\n",
      "ELF5      inf\n",
      "RAX2      inf\n",
      "FBXW12    inf\n",
      "FBXW10    inf\n",
      "FBXO47    inf\n",
      "dtype: float64\n",
      "Top 5 Protein Genes:\n",
      "Name\n",
      "FABP7      1.775598\n",
      "AKR1B10    1.454579\n",
      "MT1H       1.372425\n",
      "HMGCS2     1.246962\n",
      "MT1G       1.230909\n",
      "dtype: float64\n",
      "Combined Gene Data:\n",
      "Name                 ELF5          RAX2        FBXW12        FBXW10  \\\n",
      "Patient_ID                                                            \n",
      "C3L-00004   -7.135883e+00 -5.248972e+00 -5.720846e+00 -4.792302e+00   \n",
      "C3L-00010    1.000000e+10 -6.688864e+00  1.000000e+10 -5.647232e+00   \n",
      "C3L-00011    1.000000e+10  1.000000e+10 -5.040122e+00 -6.111578e+00   \n",
      "C3L-00026   -2.904599e+00  1.000000e+10 -4.428160e+00 -4.277224e+00   \n",
      "C3L-00079    1.000000e+10  1.000000e+10 -5.853282e+00  1.000000e+10   \n",
      "...                   ...           ...           ...           ...   \n",
      "C3N-00317.N  1.000000e+10  1.000000e+10  1.000000e+10  1.000000e+10   \n",
      "C3N-00320.N  1.000000e+10  1.000000e+10  1.000000e+10  1.000000e+10   \n",
      "C3N-00437.N  1.000000e+10  1.000000e+10  1.000000e+10  1.000000e+10   \n",
      "C3N-00492.N  1.000000e+10  1.000000e+10  1.000000e+10  1.000000e+10   \n",
      "C3N-00953.N  1.000000e+10  1.000000e+10  1.000000e+10  1.000000e+10   \n",
      "\n",
      "Name               FBXO47     FABP7   AKR1B10      MT1H    HMGCS2      MT1G  \n",
      "Patient_ID                                                                   \n",
      "C3L-00004    1.000000e+10 -1.505431 -2.447555 -2.675872 -1.981554 -2.253525  \n",
      "C3L-00010   -6.620327e+00 -1.451742 -2.618953 -4.050571 -2.761786 -4.060004  \n",
      "C3L-00011   -5.499710e+00 -1.148747 -1.053074 -1.570064 -1.488510 -1.512385  \n",
      "C3L-00026   -5.472711e+00 -1.107631 -0.950495 -2.607363  1.020557 -2.475965  \n",
      "C3L-00079   -2.727908e+00 -1.085986 -2.450258 -1.735521 -1.250407 -2.525792  \n",
      "...                   ...       ...       ...       ...       ...       ...  \n",
      "C3N-00317.N  1.000000e+10 -2.844145 -2.133574  0.291077 -0.436721  0.892795  \n",
      "C3N-00320.N  1.000000e+10 -4.757028 -2.739819  1.215191 -1.024909  1.622315  \n",
      "C3N-00437.N  1.000000e+10 -4.164262  0.144217  0.903311 -0.164857  1.260770  \n",
      "C3N-00492.N  1.000000e+10 -2.966584 -1.473006  0.981285 -0.142299  1.127847  \n",
      "C3N-00953.N  1.000000e+10 -3.272883 -1.269178  1.465206 -0.191871 -2.342646  \n",
      "\n",
      "[194 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "#Create a New Data Frame that Includes the 5 Most Differentially Expressed RNA Genes & Proteins\n",
    "top_5_rna_genes = np.abs(rna_data.mean()).sort_values(ascending = False).head(5)\n",
    "top_5_protein_genes = np.abs(protein_data.mean()).sort_values(ascending = False).head(5)\n",
    "print(\"Top 5 RNA Genes:\")\n",
    "print(top_5_rna_genes)\n",
    "print(\"Top 5 Protein Genes:\")\n",
    "print(top_5_protein_genes)\n",
    "\n",
    "combined_gene_data = pd.concat([rna_data[top_5_rna_genes.index], protein_data[top_5_protein_genes.index]], axis = 1)\n",
    "print(\"Combined Gene Data:\")\n",
    "print(combined_gene_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "13d41c2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Scaling, UMAP Embedding, & Ordinal Encoding\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from umap import UMAP\n",
    "scaler = StandardScaler()\n",
    "embedding_unscaled = UMAP().fit_transform(combined_gene_data)\n",
    "scaled_data = scaler.fit_transform(combined_gene_data)\n",
    "embedding_scaled = UMAP().fit_transform(scaled_data)\n",
    "\n",
    "from sklearn.preprocessing import OrdinalEncoder\n",
    "encoder = OrdinalEncoder()\n",
    "unencoded_columns = combined_gene_data\n",
    "encoded_columns = encoder.fit_transform(combined_gene_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "995dc856",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Compare the 4 Classification Models\n",
    "from sklearn.neighbors import KNeighborsClassifier \n",
    "from sklearn.tree import DecisionTreeClassifier\n",
    "from sklearn.neural_network import MLPClassifier\n",
    "from sklearn.naive_bayes import GaussianNB\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "classifier = KNeighborsClassifier() \n",
    "X_train, X_test, y_train, y_test = train_test_split(scaled_data, combined_gene_data, train_size = 0.7)\n",
    "classifier.fit(X_train, y_train.ravel())\n",
    "y_pred = classifier.predict(X_test)\n",
    "\n",
    "regressors_names = [\n",
    "    'KNeighborsClassifier', \n",
    "    'DecisionTreeClassifier', \n",
    "    'MLPClassifier', \n",
    "    'GaussianNB'\n",
    "]\n",
    "\n",
    "regressors = [\n",
    "    KNeighborsClassifier(),\n",
    "    DecisionTreeClassifier(),\n",
    "    MLPClassifier(),\n",
    "    GaussianNB()\n",
    "]\n",
    "\n",
    "regressors_perf = {\n",
    "    0: [],\n",
    "    1: [],\n",
    "    2: [],\n",
    "    3: []\n",
    "}\n",
    "\n",
    "for n in range(10):\n",
    "    X_train, X_test, y_train, y_test = train_test_split(scaled_data, combined_gene_data, train_size=0.7)\n",
    "    for i in range(len(regressors)):\n",
    "        model = regressors[i]\n",
    "        model.fit(X_train, y_train.ravel())\n",
    "        y_pred = model.predict(X_test)\n",
    "\n",
    "        difference = y_pred - y_test\n",
    "        regressors_perf[i].append(abs(np.mean(difference)))\n",
    "\n",
    "print('\\nAfter 10 simulations, the average error for each regressor is as follows:')\n",
    "for i in regressors_perf:\n",
    "    print(f'\\t{regressors_names[i]} : {np.mean(regressors_perf[i])}')\n",
    "    \n",
    "#DecisionTreeClassifier appears to be the best model."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
